scrapy/commands/genspider.py

```diff
@@ -85,12 +86,14 @@ def run(self, args, opts):
 
     def _genspider(self, module, name, domain, template_name, template_file):
         """Generate the spider module, based on the given template"""
+        start_url = domain if re.search('https?://', domain) else 'http://' + domain
```

Comment:
I have a doubt here,
According to the syntax of genspider ```scrapy genspider [-t template] <name> <domain>``` . We should type only domain name in place of ```<domain>```, but why are we here typing the whole URL here?

Reply:
> this still doesn't allow to pass a domain with a schema (e.g. http), because such domain would be incorrect in `allowed_domains` in the template.

@kmike Nice catch, I've pushed the fix. Thanks for reviewing.

Reply:
I second @maramsumanth question here.

Should it error in case the `domain` is an url and suggest the correct value instead of auto-fixing it ?

In the other case I think we should change genspider syntax too to make it clear it accepts an url and uses that url as start_url and extract the domain from it.

Reply:
@dangra , then it will become easy for extracting domain and start_url if we change the syntax. Also, it becomes easy for user if he/she types the whole URL instead of only domain .

Reply:
Thanks @maramsumanth @dangra for pointing this out. I'll update the documentation too.